### DISCUSSION TITLE
API get call with nested items

### DISCUSSION THREAD

**Discussion by BretOrrell** [netbox-community/netbox#18139] [Help Wanted!]
I'm trying to do an API call in /dcim/device-bays/ area. I can add a "?name=Bay1" to the search but that gives me every device with "Bay1" bays. How can I also add to search in the query to also include the nested device/name of "Test_Fiber_Box" so it only gives me this one bay?

{
            "id": 13,
            "url": "https://192.168.2.237/api/dcim/device-bays/13/",
            "display_url": "https://192.168.2.237/dcim/device-bays/13/",
            "display": "Bay1",
            "device": {
                "id": 6,
                "url": "https://192.168.2.237/api/dcim/devices/6/",
                "display": "Test_Fiber_Box",
                "name": "Test_Fiber_Box",
                "description": ""
            },
            "name": "Bay1",
            "label": "",
            "description": "",
            "installed_device": null,
            "tags": [],
            "custom_fields": {},
            "created": "2024-12-02T21:29:16.112276Z",
            "last_updated": "2024-12-02T21:29:16.112291Z"
        }

**Comment by candlerb** [comment]
Try `?name=Bay1&device_id=6` or possibly `?name=Bay1&device=Test_Fiber_Box`

**Reply by BretOrrell** [reply]
Ok.. that worked... I was thinking you had to specify the device and then the name field under it.

**Reply by BretOrrell** [reply]
I also discovered that you can use the params field in the GET string.
params = {
    'name': 'Bay3',
    'device': {'Test_Fiber_Box'}
    }

response = requests.get(NETBOX_URL, params=params, headers=netbox_headers)
